h3/src/error.rs

```diff
@@ -17,17 +17,23 @@ pub struct Error {
 
 /// An HTTP/3 "application error code".
 #[derive(PartialEq, Eq, Hash, Clone, Copy)]
-pub struct Code(u64);
+pub struct Code {
+    code: u64,
+    level: ErrorLevel,
```

Comment:
I'd probably make the error level a concept of the `Error`, and not the `Code` (I'd expect the code to be mostly only a number). Though, I'm also open to hearing why it should be part of `Code` directly...

Reply:
I did this to be able to define a specific `ErrorLevel` for a specific `Code`.
But maybe a `impl From<Code> for ErrorLevel` would also do it?

Then the `ErrorLevel` could be part of the `Application` like `Code` and `Reason`.

Reply:
After having reviewed the rest, I now lean even more that `Code` should just be the number. I noticed all the constants in this PR needed to make them connection error level codes, whereas the [spec](https://httpwg.org/specs/rfc9114.html#http-error-codes) just defines them separate. The code combined with a context turns it into a connection or stream level error.

Reply:
Hi, sorry for the late response.
I moved the `ErrorLevel` away from the `Code` struct.